library/std/src/os/horizon/thread.rs

```diff
@@ -0,0 +1,88 @@
+//! Nintendo 3DS-specific extensions to primitives in the [`std::thread`] module.
+//!
+//! All 3DS models have at least two CPU cores available to spawn threads on:
+//! The application core (appcore) and the system core (syscore). The New 3DS
+//! has an additional two cores, the first of which can also run user-created
+//! threads.
+//!
+//! Threads spawned on the appcore are cooperative rather than preemptive. This
+//! means that threads must explicitly yield control to other threads (whether
+//! via synchronization primitives or explicit calls to `yield_now`) when they
+//! are not actively performing work. Failure to do so may result in control
+//! flow being stuck in an inactive thread while the other threads are powerless
+//! to continue their work.  
+//!
+//! However, it is possible to spawn one fully preemptive thread on the syscore
+//! by using a service call (https://www.3dbrew.org/wiki/APT:SetApplicationCpuTimeLimit)
+//! to reserve a slice of time for a thread to run. Attempting to run more than
+//! one thread at a time on the syscore will result in an error.
+//!
+//! [`std::thread`]: crate::thread
+
+#![unstable(feature = "horizon_thread_ext", issue = "none")]
+
+/// Extensions on [`std::thread::Builder`] for the Nintendo 3DS.
+///
+/// [`std::thread::Builder`]: crate::thread::Builder
+pub trait ThreadBuilderExt: Sized {
```

Comment:
Didn't think of this before, but would `BuilderExt` be more consistent with `thread::Builder`? Not big deal either way I think

Reply:
I thought of that when I made the trait, but thinking about it again, I think it would work. My concern was that it would look odd as `std::os::horizon::BuilderExt`, but in fact it's `std::os::horizon::thread::BuilderExt` which looks better. I'll rename it.